models/provenance.py

```diff
@@ -294,7 +294,7 @@ def update_id(self):
         superdict = dict(
             process=self.process,
             parameters=self.parameters,
-            upstream_hashes=self.upstream_hashes,
+            upstream_hashes=self.upstream_hashes,  # this list is ordered by upstream ID
```

Comment:
Do we want to order it by upstream ID, or by the order they're stored in the index fields?

Reply:
I don't know what index fields is, in this context. 

I think the Provenance's ID is the only reasonable way to sort these. There could be many provenances (of the same type or different types) that go into this hash, and we must make sure the order of the provenances is consistent, even if they were produced in a different order (as long as they are the same set).